backend/apps/ifc_validation/checks/check_bsdd.py

```diff
@@ -1,396 +1,622 @@
+                "outcome": "Warning",
+                "message": f"Class '{class_['class_name']}' with identifier '{class_['class_identifier']}' exists in bSDD, but with name '{class_['class_bsdd_name']}'.",
+                "class": class_['class_name'],
+                "instance_id": class_['class_id']
+            }]
+
+        # Check #12 - class not in bSDD and not external --> Error
+        if not class_['class_in_bsdd'] and class_['class_identifier'] and 'buildingsmart.org' in class_['class_identifier']:
+
+            bsdd_results['messages'] += [{
+                "rule": 12,
+                "category": "class",
+                "severity": "Error",
+                "outcome": "Reference Error",
+                "message": f"Class '{class_['class_name']}' with identifier '{class_['class_identifier']}' was not found in bSDD.",
+                "class": class_['class_name'],
+                "instance_id": class_['class_id']
+            }]
+
+        # Check #13 - class not in bSDD but external --> Neutral (passed)
+        if not class_['class_in_bsdd'] and class_['class_identifier'] and not 'buildingsmart.org' in class_['class_identifier']:
+
+            bsdd_results['messages'] += [{
+                "rule": 13,
+                "category": "class",
+                "severity": "Passed",
+                "outcome": "Passed",
+                "message": f"Class '{class_['class_name']}' with identifier '{class_['class_identifier']}' was not found in bSDD, but is external.",
+                "class": class_['class_name'],
+                "instance_id": class_['class_id']
+            }]
+
+        # Check #14 - class uri does not start with https:// --> Warning 
+        if class_['class_identifier'] and not class_['class_identifier'].startswith('https://'):
+
+            bsdd_results['messages'] += [{
+                "rule": 14,
+                "category": "class",
+                "severity": "Warning",
+                "outcome": "Warning",
+                "message": f"Class identifier '{class_['class_identifier']}' does not start with https:// .",
+                "class": class_['class_name'],
+                "instance_id": class_['class_id']
+            }]
+
+        # Check #15 - class uri has dash before version 
+        # TODO
+
+    # relation constraint checks
+    for assignment in bsdd_results['assignments']:
+
+        rel_objects = assignment['assignment_rel_objects']
+        rel_class = assignment['assignment_rel_class']
+        class_ = [c for c in bsdd_results['classes'] if c['class_id'] == rel_class['id']]
+        class_ = class_[0] if len(class_) > 0 else None
+
+        # TODO - check with Artur re. IfcCommunication_s_Appliance --> confirmed, typo!
+        if class_ and 'class_bsdd_rel_objects' in class_ and class_['class_bsdd_rel_objects']:
+            for rel_object in rel_objects:
+
+                if rel_object['type'] not in class_['class_bsdd_rel_objects']:
```

Comment:
Here we should probably take into account inheritance??

E.g IfcWallStandardCase as a subtype of IfcWall?

Can you check with AT?

Reply:
Will check with Artur and open separate ticket